web/webkit/src/main/scala/net/liftweb/http/SecurityRules.scala

```diff
@@ -0,0 +1,423 @@
+ *        restrictions, though these are strongly discouraged.
+ * @param styleSources A list of source restrictions for loading styles. Also
+ *        accepts the `[[ContentSourceRestriction.UnsafeInline UnsafeInline]]`
+ *        source, though it is strongly discouraged.
+ * @param reportUri The URI where any violation of the security policy will be
+ *        reported. You can set the function that handles these violations in
+ *        `[[LiftRules.contentSecurityPolicyViolationReport]]`. By default,
+ *        reported to `[[ContentSecurityPolicy.defaultReportUri]]`.
+ *
+ *        If this is `None`, violations will not be reported.
+ */
+final case class ContentSecurityPolicy(
+  defaultSources: List[ContentSourceRestriction] = List(ContentSourceRestriction.Self),
+  connectSources: List[ContentSourceRestriction] = Nil,
+  fontSources: List[ContentSourceRestriction] = Nil,
+  frameSources: List[ContentSourceRestriction] = Nil,
+  imageSources: List[ContentSourceRestriction] = List(ContentSourceRestriction.All),
+  mediaSources: List[ContentSourceRestriction] = Nil,
+  objectSources: List[ContentSourceRestriction] = Nil,
+  scriptSources: List[JavaScriptSourceRestriction] = Nil,
+  styleSources: List[StylesheetSourceRestriction] = Nil,
+  reportUri: Option[URI] = Some(ContentSecurityPolicy.defaultReportUri)
+) {
+  /**
+   * The string that describes this content security policy in the syntax
+   * expected by the `Content-Security-Policy` header.
+   */
+  def contentSecurityPolicyString = {
+    val allRestrictions =
+      Map(
+        "default-src" -> defaultSources,
+        "connect-src" -> connectSources,
+        "font-src" -> fontSources,
+        "frame-src" -> frameSources,
+        "img-src" -> imageSources,
+        "media-src" -> mediaSources,
+        "object-src" -> objectSources,
+        "script-src" -> scriptSources,
+        "style-src" -> styleSources
+      )
+
+    val restrictionString =
+      allRestrictions
+        .collect {
+          case (category, restrictions) if restrictions.nonEmpty =>
+            category +
+              " " +
+              restrictions.map(_.sourceRestrictionString).mkString(" ")
+        }
+        .mkString("; ")
+
+    reportUri.map { uri =>
+      s"$restrictionString; report-uri $uri"
+    } getOrElse {
+      restrictionString
+    }
+  }
+
+  private lazy val reportOnlyHeaders = {
+    List(
+      "Content-Security-Policy-Report-Only" -> contentSecurityPolicyString,
+      "X-Content-Security-Policy-Report-Only" -> contentSecurityPolicyString
+    )
+  }
+  private lazy val enforcedHeaders = {
+    List(
+      "Content-Security-Policy" -> contentSecurityPolicyString,
+      "X-Content-Security-Policy" -> contentSecurityPolicyString
+    )
+  }
+  /**
+   * Returns the headers implied by this content security policy.
+   */
+  def headers(enforceInDevMode: Boolean = true): List[(String, String)] = {
+    if (! enforceInDevMode && Props.devMode) {
```

Comment:
Yeah, I get what you mean—but I think they can do that kind of customization with `supplementalHeaders` directly, since that's where these headers go. I wanted this flag specifically basically because I wanted the default dev behavior to be report-only, and the default anything-else behavior to be enforce. So I figured it deserved its own first-class flag. Willing to change it if we think there's a better way though.